src/main/java/internetshop/Factory.java

```diff
@@ -28,10 +47,7 @@
     private static UserService userServiceInstance;
 
     public static ItemDao getItemDao() {
-        if (itemDaoInstance == null) {
-            itemDaoInstance = new ItemDaoImpl();
```

Comment:
if (itemDaoInstance == null) {
            itemDaoInstance = new ItemDaoJdbcImpl(connection);
        }
        return itemDaoInstance;

Reply:
true, it's still should be a singleton